src/components/TypeRacerLevel.tsx

```diff
@@ -0,0 +1,163 @@
+              return (
+                <span style={{ position: 'absolute', width: '100%' }}>
+                  <p className="restartPrompt">
+                    Too slow...
+                    <button
+                      className="restartButton"
+                      onClick={() => {
+                        setRestart(true);
+                      }}
+                    >
+                      try again?
+                    </button>
+                  </p>
+                </span>
+              );
+            }
+            return <span className="typingTimer">:{seconds}</span>;
+          }}
+          date={Date.now() + typingTime * 1000}
+        ></Countdown>
+      )}
+      <hr className="raceTrackLine" />
+    </div>
+  );
+});
+RaceTrack.displayName = 'Race Track';
+// The text prompt. The user can change this ;)
+interface PromptProps {
+  prompt: string;
+  setPrompt: React.Dispatch<React.SetStateAction<string>>;
+}
+function TextPrompt({ prompt, setPrompt }: PromptProps): JSX.Element {
+  return (
+    <div>
+      <form>
+        <textarea
+          value={prompt}
+          onChange={(e) => setPrompt(e.target.value)}
+          className="textPrompt"
+        />
+      </form>
+    </div>
+  );
+}
+
+function TypeRacerLevel(props: LevelProps): JSX.Element {
+  const [redirect, setRedirect] = useState(false);
+  const [input, setInput] = useState('');
+  const p = prompts[Math.floor(Math.random() * prompts.length)]; // Triggered weird comma dangle error
+  const [prompt, setPrompt] = useState(p);
+  const [restart, setRestart] = useState(true);
+  const [timeOut, setTimeOut] = useState(true);
+  // True at the first load up and when user clicks restart button after losing
+  if (restart) {
+    return (
+      <div>
+        <p style={{ textAlign: 'center' }}>Type Racing</p>
+        <div className="raceTrackContainer">
+          <RaceTrack />
+          <div>
+            <p className="restartContainer">Starting in{'      '}</p>
+            <Countdown
+              renderer={({ seconds, completed }) => {
+                if (completed) {
+                  const p2 =
+                    prompts[Math.floor(Math.random() * prompts.length)];
+                  setInput('');
+                  setPrompt(p2);
+                  setRestart(false);
+                  setTimeOut(false);
+                }
+                return <span className="restartTimer">:{seconds}</span>;
+              }}
+              date={Date.now() + startingTime * 1000}
+            ></Countdown>
+          </div>
+        </div>
+      </div>
+    );
+  }
+  // After finishing the game, there is a 3 second delay before going to the next level.
+  if (redirect) {
+    return <Redirect to={props.nextLevel} />;
+  }
+  // User beat the game
+  else if (input === prompt && !timeOut) {
+    setTimeout(() => setRedirect(true), 3000);
+    return (
+      <div>
+        <p style={{ textAlign: 'center' }}>Type Racing</p>
```

Comment:
i see this `style={{ textAlign: 'center' }}` a lot.. thoughts about making this a css class?

@UCLA-Creative-Labs/tech would like other ppl's opinion on this too